src/dbnode/encoding/iterators.go

```diff
@@ -38,21 +39,71 @@ var (
 // iterators is a collection of iterators, and allows for reading in order values
 // from the underlying iterators that are separately in order themselves.
 type iterators struct {
-	values      []Iterator
-	earliest    Iterator
-	earliestIdx int
-	earliestAt  time.Time
-	filtering   bool
-	filterStart time.Time
-	filterEnd   time.Time
+	values             []Iterator
+	earliest           []Iterator
```

Comment:
Can I convince you to write a little blurb about the intent behind how `earliest` and `values` are used in the code? I think I follow based on the implementation of `moveToValidNext()` but it's a little convoluted the first time someone looks at it.

Reply:
Sure thing, can do.